### PR TITLE
Change TextInput prop check to print instead of throw

### PR DESCRIPTION
**Pull request by dozoisch**
Follow up of https://github.com/facebook/react-native/pull/8499! Open to discussion

Motivation: 

See https://github.com/facebook/react-native/pull/8499#issuecomment-230004068

Goal is to have the same failure mode in dev and prod.

/cc @ide

### CODE REVIEW

**Comment by facebook-github-bot-3** [comment]
By analyzing the blame information on this pull request, we identified @JoelMarcey and @Bhullnatik to be potential reviewers.

File: Libraries/Components/TextInput/TextInput.js [hunk 1]
```diff
@@ -550,9 +550,9 @@ const TextInput = React.createClass({
       if (__DEV__) {
         for (var propKey in onlyMultiline) {
           if (props[propKey]) {
-            throw new Error(
+            console.error(new Error(
```
Comment (ide) [inline Libraries/Components/TextInput/TextInput.js]: Use the same warning method that React uses:

```js
warning(false, '%s', error.stack);
```

**Comment by ide** [comment]
Thanks for following up!

**Comment by dozoisch** [comment]
@ide Ok, I've updated it!

**Comment by ide** [comment]
@facebook-github-bot shipit

**Comment by facebook-github-bot-2** [comment]
Thanks for importing. If you are an FB employee go to [Phabricator](https://our.intern.facebook.com/intern/opensource/github/pull_request/825974120866295/int_phab) to review.

**Pull request closed without merging**